tapis-project/tapis-ui
# task/TUI-47 - Redux FileOperationsApi

## Overview: ##

`useFiles` service and temporary components

## Related Github Issues: ##

* [TUI-47](https://github.com/tapis-project/tapis-ui/issues/47)

## Summary of Changes: ##

- `useFiles` hook with a `list` operation and a cached state for file listings
- A temporary `FileListing.tsx` React component
- Changes to `apiSaga` that specify `onRequest`, `onSuccess` and `onFailure` callbacks that generate dispatches (this replaces the fixed `dispatches` object)
- A `TapisListingResults` generic type and utility functions that allow ease of updating list results that have offsets and limits (to be used when implementing pagination for long listings)
- Updated `useSystems` hooks to conform to the above changes

## Testing Steps: ##
1. npm install
2. Connect to VPN
3. npm run dev
4. Log in as testuser8/testuser8
5. Click on Systems in the sidebar
6. Click on testuser8-e2e (there will not be UI feedback but you can see console output). This will select testuser8-e2e as the system for getting file listings
7. Click on Files in the sidebar. You should see file listings

## UI Photos:

## Notes: ##

A wiki describes all of the changes and the developer workflow

https://github.com/tapis-project/tapis-ui/wiki/Tapis-Redux-Developer-Guide
